### PR TITLE
chore: [IOCOM-1742,IOCOM-1756] Disabled notifications banner in profile/settings home 

### PR DESCRIPTION
**Pull request by forrest57**
## Short description
addition of said banner, only shown when notification permissions are not granted to the app
Screenshot 2024-10-08 at 15 58 15

## List of changes proposed in this pull request
- banner
- required I18n
- selector for correct banner order rendering
- current banner component has been extrapolated in a separate one, in order to declutter the main screen

## How to test
navigate to the profile screen, and check that:
- with notifications disabled, the red banner is shown
- with notifications enabled, the fiscalCode one is shown, unless previously closed
- if previously closed, no banner is shown
- the notification banner opens the system's settings for the app, in the correct "notifications" section.

### CODE REVIEW

**Review by Vangaorth — CHANGES_REQUESTED** [review]
> [...] tests and A11y coming in next PR [...]

Please add automatic tests in this PR

File: ts/screens/profile/ProfileMainScreenTopBanner.tsx [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+import { useIONavigation } from "../../navigation/params/AppParamsList";
+import ROUTES from "../../navigation/routes";
+import { useIODispatch, useIOSelector } from "../../store/hooks";
+
+export const ProfileMainScreenTopBanner = () => {
+  const bannerToShow = useIOSelector(profileBannerToShowSelector);
+  const dispatch = useIODispatch();
+  const navigation = useIONavigation();
+
+  const handleCloseBanner = () => dispatch(setShowProfileBanner(false));
```
Comment (Ladirico) [inline ts/screens/profile/ProfileMainScreenTopBanner.tsx:21]: In order to optimize code you can use useCallback 
```suggestion
  const handleCloseBanner = useCallback(
    () => dispatch(setShowProfileBanner(false)),
    [dispatch]
  );
```

File: ts/screens/profile/ProfileMainScreenTopBanner.tsx [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  const navigateToProfile = () =>
+    navigation.navigate(ROUTES.PROFILE_NAVIGATOR, {
+      screen: ROUTES.PROFILE_DATA
+    });
```
Comment (Ladirico) [inline ts/screens/profile/ProfileMainScreenTopBanner.tsx:25]: as above

```suggestion
  const navigateToProfile = useCallback(
    () =>
      navigation.navigate(ROUTES.PROFILE_NAVIGATOR, {
        screen: ROUTES.PROFILE_DATA
      }),
    [navigation]
  );
```

Comment (forrest57) [reply ts/screens/profile/ProfileMainScreenTopBanner.tsx]: updated!

File: ts/screens/profile/__test__/ProfileMainScreenTopBanner.test.tsx [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+      jest
+        .spyOn(profileBannerImport, "profileBannerToShowSelector")
+        .mockImplementation((_: GlobalState) => selectorValue);
+
+      const component = renderComponent(<ProfileMainScreenTopBanner />);
+
+      expect(component).toMatchSnapshot(description);
+    });
+  });
+  it("should only render the notificationr banner", () => {
```
Comment (Vangaorth) [inline ts/screens/profile/__test__/ProfileMainScreenTopBanner.test.tsx:46]: This test is already covered by the snapshot test above

File: ts/screens/profile/__test__/ProfileMainScreenTopBanner.test.tsx [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+    jest
+      .spyOn(profileBannerImport, "profileBannerToShowSelector")
+      .mockImplementation((_: GlobalState) => "NOTIFICATIONS");
+
+    const component = renderComponent(<ProfileMainScreenTopBanner />);
+    expect(component.findByTestId("notifications-banner")).toBeTruthy();
+    expect(component.queryByTestId("fiscal-code-banner")).toBeNull();
+  });
+  it("should only render the fiscalCode banner", () => {
```
Comment (Vangaorth) [inline ts/screens/profile/__test__/ProfileMainScreenTopBanner.test.tsx:55]: This test is already covered by the snapshot test above

File: ts/screens/profile/__test__/ProfileMainScreenTopBanner.test.tsx
Comment (Vangaorth) [inline ts/screens/profile/__test__/ProfileMainScreenTopBanner.test.tsx:1]: We are missing three tests:
- tap on the notification alert's CTA should invoke `openNotificationSettingsScreen`
- tap on the profile banner's CTA should invoke `navigate`, with two parameters, the first equal to `ROUTES.PROFILE_NAVIGATOR` and the second equal to `{ screen: ROUTES.PROFILE_DATA }`
- tap on the profile banner's close button should invoke `dispatch` with one parameter equal to `setShowProfileBanner(false)`

File: ts/screens/profile/__test__/ProfileMainScreenTopBanner.test.tsx [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+      }
+    ] as const;
+    testCases.forEach(({ selectorValue, description }) => {
+      jest
+        .spyOn(profileBannerImport, "profileBannerToShowSelector")
+        .mockImplementation((_: GlobalState) => selectorValue);
+
+      const component = renderComponent(<ProfileMainScreenTopBanner />);
+
+      expect(component).toMatchSnapshot(description);
```
Comment (Vangaorth) [inline ts/screens/profile/__test__/ProfileMainScreenTopBanner.test.tsx:43]: Why don't we do `expect(component.toJSON())`?

**Review by Vangaorth — CHANGES_REQUESTED** [review]

File: ts/features/profileSettings/store/selectors/index.ts [hunk 6]
```diff
@@ -1,7 +1,19 @@
 import { GlobalState } from "../../../../store/reducers/types";
+import { areNotificationPermissionsEnabled } from "../../../pushNotifications/store/reducers/environment";
 
 export const showProfileBannerSelector = (state: GlobalState) =>
   state.features.profileSettings.showProfileBanner;
 
+export const profileBannerToShowSelector = (state: GlobalState) => {
```
[the remark this answers is not in the document]
Comment (forrest57) [reply ts/features/profileSettings/store/selectors/index.ts:7]: not a huge change but why not?

File: ts/screens/profile/ProfileMainScreenTopBanner.tsx [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+
+  const navigateToProfile = useCallback(
+    () =>
+      navigation.navigate(ROUTES.PROFILE_NAVIGATOR, {
+        screen: ROUTES.PROFILE_DATA
+      }),
+    [navigation]
+  );
+
+  switch (bannerToShow) {
```
[the remark this answers is not in the document]
Comment (forrest57) [reply ts/screens/profile/ProfileMainScreenTopBanner.tsx:33]: while it does look cleaner as of right now, I believe the weight added by the small duplication we have in the current switch-case allows for easier customization and readability, while also not adding any performance cost that is worth noting -- what do you think?

File: ts/screens/profile/__test__/ProfileMainScreenTopBanner.test.tsx [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+    "@react-navigation/native"
+  ),
+  useNavigation: () => ({
+    navigate: mockNavigate
+  })
+}));
+
+describe("ProfileMainScreenTopBanner", () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
```
[the remark this answers is not in the document]
Comment (forrest57) [reply ts/screens/profile/__test__/ProfileMainScreenTopBanner.test.tsx:19]: I believe it does not add anything, but why not?

File: ts/screens/profile/__test__/ProfileMainScreenTopBanner.test.tsx (hunk 5 above)
Comment (forrest57) [reply ts/screens/profile/__test__/ProfileMainScreenTopBanner.test.tsx]: didn't see the need -- added!

**Review by Vangaorth — APPROVED** [review]
LGTM

**Pull request merged**